kafka-rest/src/test/java/io/confluent/kafkarest/controllers/PartitionManagerImplTest.java

```diff
@@ -370,6 +371,28 @@ public void getPartition_nonExistingCluster_throwsNotFound() throws Exception {
     }
   }
 
+  @Test
+  public void getPartition_nonExistingCluster_throwsUnknownTopicOrPartition() throws Exception {
+    expect(topicManager.getTopic(CLUSTER_ID, TOPIC_NAME))
+        .andReturn(
+            failedFuture(
+                new UnknownTopicOrPartitionException(
+                    String.format(
+                        "This server does not host topic-partition %d for topic %s",
```

Comment:
Better to have a different message here to verify that we are correctly re-wrapping the exception with the message provided in your `PartitionManagerImpl` code.